pkg/services/services.go

```diff
@@ -96,3 +111,89 @@ func kubeProxyConfiguration(debug bool) servicescm.Service {
 		Priority:     2,
 	}
 }
+
+// getKubeletServiceConfiguration returns the Service definition for the kubelet
+func getKubeletServiceConfiguration(argsFromIginition map[string]string, debug bool, platform string) (servicescm.Service, error) {
+	kubeletArgs, err := generateKubeletArgs(argsFromIginition, debug)
+	if err != nil {
+		return servicescm.Service{}, err
+	}
+	var powershellVars []servicescm.PowershellCmdArg
+	hostnameArg := getKubeletHostnameOverride(platform)
+	if hostnameArg != "" {
+		kubeletArgs = append(kubeletArgs, hostnameArg)
+		powershellVars = append(powershellVars, servicescm.PowershellCmdArg{
+			Name: ec2HostnameVar,
+			Path: "Get-EC2InstanceMetadata -Category LocalHostname",
```

Comment:
@sebsoto, with the need for the hostname override in GCP, I feel there is a disconnection here between `getKubeletHostnameOverride()`  and the way the variable is getting resolved.  `Get-EC2InstanceMetadata` is platform specific and will only work in AWS. 

Can this be done in a way that could be extended based on the platform type?

Reply:
The `Get-EC2InstanceMetadata` is part of the [AWS.Tools for PowerShell](https://docs.aws.amazon.com/powershell/latest/userguide/pstools-welcome.html), and it comes pre-installed on Windows machines. What about BYOH instances in AWS, will the AWS.Tools for PowerShell be available?